### PR TITLE
Updated QuoteSourceVec to match F::Q v1.59

### PR DESCRIPTION
**Pull request by bpschuck**
Updated libgnucash/engine/gnc-commodity.cpp removing non-existent and non-working modules from the `QuoteSourceVec single_quote_sources` and `QuoteSourceVec multiple_quote_sources` JSON structures.

I searched for bugs related to what is available in the Security Editor, but only found [799028](https://bugs.gnucash.org/show_bug.cgi?id=799028) where the reporter asked to have `yahooweb` added as an alternative to `yahoo_json`.

### CODE REVIEW

**Review by fellen — COMMENTED** [review]
This is a good starting point!
But a few questions rise for me:

1.  Can we also get the new sources?
2. Can we also get an update of the docs ([source](https://github.com/Gnucash/gnucash-docs/blob/stable/C/manual/tips-appendix.docbook), [result](https://code.gnucash.org/docs/C/gnucash-manual/tips.html))? Targets are to help users to find
2.1. a replacement for an obsolete source;
2.2. a source for a new commodity.

**Comment by bpschuck** [comment]
On 4/14/24 3:51 PM, Frank H. Ellenberger wrote:

>  1. Can we also get the new sources?
>  2. Can we also get an update of the docs (source
>     <https://github.com/Gnucash/gnucash-docs/blob/stable/C/manual/tips-appendix.docbook>, result <https://code.gnucash.org/docs/C/gnucash-manual/tips.html>)? Targets are to help users to find
>     2.1. a replacement for an obsolete source;
>     2.2. a source for a new commodity.

Hello Frank,

I was not aware of these. I'll take a look and see what I can do. At the 
very least removing references to modules/methods that no longer exist 
and/or work. Looks like it hasn't been updated since F::Q v1.47 so I 
guess any update is better than none.

Thanks.

Bruce S

**Comment by jralls** [comment]
@bpschuck It looks to me that this is actually up to date for 1.60 as your second commit removes `fool` and `cdbfundlibrary` and `fundata` weren't there in the first place. Any other mods needed?

**Comment by jralls** [comment]
>  Can we also get the new sources?

There's a long list still showing up in "unknown": asegr, aufunds, bamosz, bams, bet, bloomberg,…. `gnucash-cli -Q info` reports 78 sources, but there are only 44 identified in gnc-commodity.cpp after this commit.

Of those first 6, 3 are *removed* by this PR: asegr, bamosz, and bet. Huh?

**Comment by bpschuck** [comment]
On 4/20/24 5:33 PM, John Ralls wrote:

> Can we also get the new sources?

Not exactly sure what you mean buy this question?

> There's a long list still showing up in "unknown": asegr, aufunds, 
> bamosz, bams, bet, bloomberg,…. |gnucash-cli -Q info| reports 78 
> sources, but there are only 44 identified in gnc-commodity.cpp after
>  this commit.

> Of those first 6, 3 are /removed/ by this PR: asegr, bamosz, and bet.
> Huh?

There are just 47 modules that have retrieval methods. What exactly is 
|gnucash-cli -Q info| returning? Both single and multiple (nyse, nasdaq, 
usa, etc) methods? aka the count of methods returned by the older 
|gnc-fq-check|?

Until the recent YahooJSON fiasco I was going to go back to this and go 
over it, as well as take a stab at the documentation pages Frank E. 
pointed me to.

Bruce S.

**Comment by jralls** [comment]
```
Application Path /Applications/Gnucash.app/Contents/MacOS/gnucash-cli
Found Finance::Quote version 1.61.
Finance::Quote sources:
aex          alphavantage              amfiindia    asegr        asx
aufunds      australia    bamosz       bats         bet          bloomberg
bourso       bse          bseindia     bvb          canada       comdirect
consorsbank  cse          deka         dutch        dwsfunds     europe
fetch_live_currencies     finanzpartner             fondsweb     france
ftfunds      goldmoney    googleweb    greece       hu           hufund
hungary      hustock      iexcloud     india        indiamutual
known_currencies          marketwatch  morningstarau
morningstarch             morningstarjp             morningstaruk
mstaruk      nasdaq       nseindia     nyse         nzx          onvista
oslobors     poland       romania      seb_funds    sinvestor    six
stooq        tesouro_direto            tiaacref     tmx          tradegate
tradeville   treasurydirect            troweprice   troweprice_direct
tsp          tsx          twelvedata   ukfunds      unionfunds   usa
xetra        yahoo_json   yahoojson    yahooweb     za
```
by running
```
sub print_version  {
    my $quoter = Finance::Quote->new();
    my @sources = $quoter->sources();
    print "$Finance::Quote::VERSION\n";
    foreach my $source (@sources) {
        print "$source\n";
    }
    exit 0;
}
```

**Comment by bpschuck** [comment]
John,

Ah, now I remember...

I also did not include modules that were indicated to be failing or 
otherwise not working in the Modules-README.yml file in the F::Q repo. 
There are a few that are indicated as failing, but still exist as part 
of the F::Q package. ASEGR.pm is one of those.

This is likely the cause of the discrepancy you pointed out.

I figured if anything, the list in my PR was far more up to date than 
the list of methods from what looks like F::Q v1.48...

Sorry about that. There was a plan in my head to go through these and do 
come cleanup work for the next release. But that release and the next 
(1.60 and 1.61) were sort of rushed out the door.

File: libgnucash/engine/gnc-commodity.cpp [hunk 1]
```diff
@@ -189,65 +189,33 @@ static QuoteSourceVec single_quote_sources =
...
-    { false, SOURCE_SINGLE, "Morningstar, SE", "morningstar" },
-    { false, SOURCE_SINGLE, "Motley Fool, US", "fool" },
     { false, SOURCE_SINGLE, "New Zealand stock eXchange, NZ", "nzx" },
     { false, SOURCE_SINGLE, "Paris Stock Exchange/Boursorama, FR", "bourso" },
-    { false, SOURCE_SINGLE, "Paris Stock Exchange/LeRevenu, FR", "lerevenu" },
-    { false, SOURCE_SINGLE, "Platinum Asset Management, AU", "platinum" },
     { false, SOURCE_SINGLE, "Romania", "romania" },
-    { false, SOURCE_SINGLE, "SIX Swiss Exchange funds, CH", "sixfunds" },
-    { false, SOURCE_SINGLE, "SIX Swiss Exchange shares, CH", "sixshares" },
+    { false, SOURCE_SINGLE, "SIX Swiss Exchange shares, CH", "six" },
```
Comment (fellen) [inline libgnucash/engine/gnc-commodity.cpp:208]: Funds no longer available?

File: libgnucash/engine/gnc-commodity.cpp [hunk 2]
```diff
@@ -189,65 +189,33 @@ static QuoteSourceVec single_quote_sources =
...
-    { false, SOURCE_SINGLE, "Greece", "greece" },
-    { false, SOURCE_SINGLE, "Helsinki stock eXchange, FI", "hex" },
-    { false, SOURCE_SINGLE, "Hungary", "hu" },
+    { false, SOURCE_SINGLE, "Google Web, US Stocks", "googleweb" },
     { false, SOURCE_SINGLE, "India Mutual", "indiamutual" },
-    { false, SOURCE_SINGLE, "Man Investments, AU", "maninv" },
-    { false, SOURCE_SINGLE, "Morningstar, GB", "mstaruk" },
+    { false, SOURCE_SINGLE, "Morningstar, GB", "morningstaruk" },
     { false, SOURCE_SINGLE, "Morningstar, JP", "morningstarjp" },
-    { false, SOURCE_SINGLE, "Morningstar, SE", "morningstar" },
```
Comment (fellen) [inline libgnucash/engine/gnc-commodity.cpp:223]: AFAIK they are rating funds.
Offer both the same funds or a  regional subset?

**Comment by jralls** [comment]
@bpschuck 

>Sorry about that. There was a plan in my head to go through these and do some cleanup work for the next release.

So cleanup first and remove the broken sources from GnuCash second. GnuCash parses the list from `quoter->sources()`, disables the hard-coded entries that aren't listed, and adds at the bottom of the GUI list under the heading "Unknown" the listed items that don't have hard-coded entries. Removing hard-coded entries that are reported by `sources` just moves the sources to the bottom.

There are a couple of holes in that system: First, the code that creates the "unknown" list runs in the GUI because it's a GUI display list so the "unknown" sources [aren't available from gnucash-cli](https://bugs.gnucash.org/show_bug.cgi?id=799066). Perhaps worse there's no handling for obsolete or erroneous sources in a user's book. This is [implicated in a crash](https://bugs.gnucash.org/show_bug.cgi?id=799285). To be clear I don't expect you to fix either of those problems.

**Comment by jralls** [comment]
I managed to commit this after a rebase so it's in stable.

**Comment by bpschuck** [comment]
Cool. I had thought you were waiting on me to do some more F::Q cleanup and the doc updates Frank had suggested.
Glad I was able to contribute at least a tiny bit.

**Pull request closed without merging**